server/src/main/java/org/opensearch/index/mapper/RewriteOverride.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.mapper;
+
+import org.opensearch.common.annotation.PublicApi;
+
+/** A custom rewrite override for a query. Default executes the query as is and other values determine which structure to use at run-time.
+*
+* @opensearch.internal
+*/
+@PublicApi(since = "2.17.0")
+public enum RewriteOverride {
+
+    // don't override the rewrite, use default
```

Comment:
I think the default is not informative, may be:

```suggestion
    INDEX_OR_DOC_VALUES
```

Reply:
updated